### PR TITLE
[DEN-294] Added bot rules to SDK

### PR DESCRIPTION
**Pull request by stevenpaz**

### CODE REVIEW

File: example/waf/bot_rules/bot_rules.go [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+	"github.com/EdgeCast/ec-sdk-go/edgecast/waf"
+)
+
+// Demonstrates the usage of WAF Bot Rules
+//
+// Usage:
+// go run bot_rules.go
+//
+// For detailed information about Bot Rules in WAF, please refer to:
+// TODO [URL TBD]
```
Comment (stevenpaz) [inline example/waf/bot_rules/bot_rules.go:19]: Not sure if we should wait for the official documentation to be up or just leave this blank

File: edgecast/waf/bot_rule_models.go [hunk 2]
```diff
@@ -0,0 +1,111 @@
+// Copyright 2022 Edgecast Inc., Licensed under the terms of the Apache 2.0
+// license. See LICENSE file in project root for terms.
+
+package waf
+
+// BotRule is a detailed representation of a Bot Manager Rule
+type BotRule struct {
+	/*
```
Comment (wdawg33) [inline edgecast/waf/bot_rule_models.go:8]: We generally use line comments within the struct when defining fields. Should we discuss using block comments further as a team?

File: edgecast/waf/bot_rule_models.go [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+	return GetBotRuleParams{}
+}
+
+// GetBotRuleParams -
+type GetBotRuleParams struct {
+	AccountNumber string
+	BotRuleID     string
+}
+
+// BotRuleGetOK -
```
Comment (wdawg33) [inline edgecast/waf/bot_rule_models.go:57]: Missing description or are we waiting for the API documentation to grab it from there?

File: edgecast/waf/bot_rule_models.go [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+	return AddBotRuleParams{}
+}
+
+// AddBotRuleParams -
+type AddBotRuleParams struct {
+	BotRule       BotRule
+	AccountNumber string
+}
+
+// BotRuleAddOK -
```
Comment (wdawg33) [inline edgecast/waf/bot_rule_models.go:87]: Missing description or are we waiting for the API documentation to grab it from there?

File: example/waf/bot_rules/bot_rules.go (hunk 1 above)
Comment (wdawg33) [reply example/waf/bot_rules/bot_rules.go:19]: The TODO is okay if we create a followup ticket.

**Review by wdawg33 — CHANGES_REQUESTED** [review]

File: edgecast/waf/bot_rule_models.go (hunk 2 above)
Comment (stevenpaz) [reply edgecast/waf/bot_rule_models.go:8]: I use block comments for lenghty descriptions. They seem to allow some basic formatting for intellisense as well.

File: example/waf/bot_rules/bot_rules.go (hunk 1 above)
Comment (stevenpaz) [reply example/waf/bot_rules/bot_rules.go]: URL has been updated

File: edgecast/waf/bot_rule_models.go (hunk 2 above)
Comment (stevenpaz) [reply edgecast/waf/bot_rule_models.go]: This was an incorrect statement about formatting. The comments have been converted to line comments.

File: edgecast/waf/bot_rule_models.go (hunk 3 above)
Comment (stevenpaz) [reply edgecast/waf/bot_rule_models.go]: Descriptions have been updated

File: edgecast/waf/bot_rule_models.go (hunk 4 above)
Comment (stevenpaz) [reply edgecast/waf/bot_rule_models.go]: Descriptions have been updated

**Review by defstream — CHANGES_REQUESTED** [review]
My only comment is concerning Valid Values...
With a specific set of possible values, it is better to turn them into an enum that serializes to a string; below is an example.

File: edgecast/waf/common.go [hunk 5]
```diff
@@ -56,3 +56,264 @@ type WAFError struct {
...
+			REQUEST_HEADERS |
+			REQUEST_METHOD |
+			REQUEST_URI
+
+		Note: If a request element consists of one or more key-value pairs,
+		then you may identify a key via a match object.
+		If is_count has been disabled, then you may identify a specific
+		value via the operator object.
+	*/
+	Type string `json:"type"`
```
[the remark this answers is not in the document]
Comment (stevenpaz) [reply edgecast/waf/common.go:168]: Ah that's a good point. Lemme see if can whip that up real quick.

**Review by wdawg33 — APPROVED** [review]

File: edgecast/waf/common.go (hunk 5 above)
[the remark this answers is not in the document]
Comment (stevenpaz) [reply edgecast/waf/common.go]: @defstream updated, let me know what you think.

**Review by defstream — APPROVED** [review]

**Review by defstream — APPROVED** [review]

**Pull request merged**